src/env.rs

```diff
@@ -82,7 +123,12 @@ impl Source for Environment {
             .as_ref()
             .map(|prefix| format!("{}{}", prefix, group_separator).to_lowercase());
 
-        for (key, value) in env::vars() {
+        let source: Box<dyn Iterator<Item = (String, String)>> = match &self.source {
+            Some(source) => Box::new(source.clone().into_iter()),
+            None => Box::new(env::vars()),
+        };
+
+        for (key, value) in source {
```

Comment:
I don't like this. I think it would be better to move the body of the for loop into a closure, then in the match you'd do something like this instead:

```rust
match &self.source {
    Some(source) => source.cloned().into_iter().for_each(body),
    None => env::vars().for_each(body),
}
```

Reply:
You are right, that looks way cleaner. Thanks for sharing the improvement!